### ISSUE TITLE
Running full MVC site in Lambda

### ISSUE THREAD

**Issue by Kralizek** [aws/aws-lambda-dotnet#37]
Hi,

during re:Invent 2016, in the session DEV 207, we saw we could host an "ASP.NET Core" API application in Lambda/API Gateway.

Is there any specific reason why the sample showed only an API endpoint?

The reason I'm asking is because I'm wondering the feasibility of hosting a traditional website in Lambda (including access to an SQL Server instance on RDS.

I'm not sure if this is the correct place to formulate the question,

**Comment by normj** [comment]
Hello

I have focused with Web API because they seemed like a natural thing to run on Lambda and API Gateway. As far as running a full site I think most things would probably work. I noticed when I tried it that the compilation of the razor pages isn't working which I plan on looking more into.

Static content like images and Javascript being served by a Lambda function seems like overkill but I know it works. Another concern is the entire static content being requested would get loaded into memory before passing onto API Gateway so that could force you to set high memory requirement on the Lambda function. I think a more ideal deployment would be to deploy the static content to S3 and then the code to Lambda. There is trickiness in how you configure the API Gateway for that but that is something I want to research to see if we can make easier with tooling and a common project layout.
